docs/en-us/dev/user_doc/alert_spi.md

```diff
@@ -0,0 +1,102 @@
+
+
+
+#### Alert SPI Main class information.
+
+AlertChannelFactory
+AlertChannelFactory, all alert plugins need to implement this interface, the interface is used to define the name of the alert plugin, the required parameters, the create method is used to create a specific instance of the alert plugin.
+
+AlertChannel
+The interface of the alarm plug-in, the alarm plugin needs to implement the interface, only one method process, the upper alarm system calls the method and returned by the AlertResult to get the return information of the alert.
+
+AlertData
+Alert content information, including id, title, content, log.
+
+AlertInfo
+Alarm related information, when the upper system calls the alarm plug-in instance, the instance of this class is incorporated into the specific alarm plugin through the Process method. The internal contains the parameter information filled in the front end of the alarm content AlertData and the called alarm plugin instance.
+
+AlertResult
+The alert plugin sends an alert return message.
+
+org.apache.dolphinscheduler.spi.params 
+This package contains the plug-in parameter definitions. We use alpacajs, a front-end library http://www.form-create.com, which dynamically generates the front-end ui based on the parameter list json returned by the plug-in definition, so we don't need to care about the front-end when doing SPI plug-in development.
+
+Under this package we currently only have RadioParam, TextParam and PasswordParam, which are used to define parameters of text, radio and password respectively.
+
+AbsPluginParams This class is the base class for all parameters and is inherited by the RadioParam classes. Each DS alert plugin returns a list of AbsPluginParams in the AlertChannelFactory implementation.
+
+The design of alert_spi can be found in issue: [Alert Plugin Design](https://github.com/apache/incubator-dolphinscheduler/issues/3049)
+
+#### Alert SPI built-in implementation
+
+* Email
+
+     Email alert notification
+
+* DingTalk
+
+     Alert for DingTalk group chat bots
+
+* EnterpriseWeChat
+
+     EnterpriseWeChat alert notifications
+
+     Related parameter configuration can refer to the EnterpriseWeChat robot document.
+
+* Script
+
+     We have implemented a shell script for alerting, we will pass the relevant alert parameters to the script and you can implement your alert logic in the shell, this is a good way to interface with internal alerting applications.
+
+* SMS
+
+     SMS alerts
+
+#### Alarm Custom Plugin Development
+
+In fact, it's very easy to implement a plugin  by yourself, you only need to care about the plugin extension interface, in Alert you only need to care about the AlertChannelFactory and AlertChannel. We would recommend that you follow the design specifications of other built-in plugins so that when your idea is good enough, you can donate it to the community without having to change it too much.

```

Comment:
Two spaces between "plugin by".
Changing the comma before "in Alert" to an end stop.